### ISSUE TITLE
Manual : add groovy DSL examples alongside java examples

### ISSUE THREAD

**Issue by javazquez** [ratpack/ratpack#776]
I think it would be really great to see the equivalent groovy DSL code along side the java examples in the manual. My first thought would be a tab that a user could toggle with a label of "java" or "groovy" to see the example code for each code example provided.

**Comment by adeady** [comment]
I want to second this.  I have created a project using lazy-bones and groovy (current status can be found [here](https://github.com/adeady/lazy-rat)).  I want to use spring-boot with RatPack because I am familiar with many of the libraries, but the examples on the web for groovy+spring in RatPack are out of date.  Specifically, the issues I am running into is accessing a spring data repository --or any spring bean-- in the `Ratpack.groovy` file.  I have seen examples [in java](http://www.infoq.com/articles/Ratpack-and-Spring-Boot#anch126182) but I would rather use the groovy dsl.  I was following [this](https://objectpartners.com/2015/05/12/my-first-ratpack-app-what-i-learned/) guide for using the groovy dsl, but each time they pull in a module they use a `handler` closure which each time I add a 
```
handlers {
        handler('bob') {
            println "foo"
        }
        ...
}
```
I get the following error:
```
groovy.lang.MissingMethodException: No signature of method: ratpack.handler() is applicable for argument types: (java.lang.String, ratpack$_run_closure1$_closure3$_closure4) values: [bob, ratpack$_run_closure1$_closure3$_closure4@41343f]
```

I have also seen a lot of [examples](https://gist.github.com/beckje01/136fce42cdba7036a988) using a `Registries` object which appears to now be [refactored](https://github.com/ratpack/ratpack/issues/710).  This is causing me issues following previous tutorials and how-tos since I do not know which parts of the API are still relevant. 

TL;DR:  The groovy dsl needs more documentation around how to use a registry and bindings in Ratpack.groovy and there needs to be up to date instructions for integrating spring boot with RatPack.

**Comment by rhart** [comment]
@adeady documentation is our next big focus area. Better sign posting to what we already have and filling in any gaps.  We need to decide whether we're going to provide examples in both Java and Groovy but this is also some feedback I've had at a workshop I did.

For Groovy examples the tests are actually quite a good reference. Meanwhile I will send you a PR for what you are trying to do.

**Comment by alkemist** [comment]
While we'd like to do this… it's just too much overhead to manage sorry.
